src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -813,72 +811,72 @@ public string GetDisplayName(SyntaxNode? node, DisplayNameOptions options, strin
+                    AppendTypeParameterList(builder, memberDeclaration.GetTypeParameterList());
+
+                return builder.ToString();
             }
-            else
+            else if (memberDeclaration is ExtensionDeclarationSyntax extensionDeclaration)
             {
-                var nameToken = memberDeclaration.GetNameToken();
-                if (nameToken != default)
-                {
-                    name = nameToken.IsMissing ? missingTokenPlaceholder : nameToken.Text;
-                    if (memberDeclaration.Kind() == SyntaxKind.DestructorDeclaration)
-                    {
-                        name = "~" + name;
-                    }
+                using var _ = PooledStringBuilder.GetInstance(out var builder);
+                builder.Append("extension");
 
-                    if ((options & DisplayNameOptions.IncludeTypeParameters) != 0)
-                    {
-                        var pooled = PooledStringBuilder.GetInstance();
-                        var builder = pooled.Builder;
-                        builder.Append(name);
-                        AppendTypeParameterList(builder, memberDeclaration.GetTypeParameterList());
-                        name = pooled.ToStringAndFree();
-                    }
-                }
-                else
-                {
-                    Debug.Assert(memberDeclaration.Kind() == SyntaxKind.IncompleteMember);
-                    name = "?";
-                }
+                if ((options & DisplayNameOptions.IncludeTypeParameters) != 0)
+                    AppendTypeParameterList(builder, memberDeclaration.GetTypeParameterList());
+
+                AppendParameterList(builder, extensionDeclaration.ParameterList);
+                return builder.ToString();
             }
-        }
-        else
-        {
-            if (node is VariableDeclaratorSyntax fieldDeclarator)
+            else
             {
-                var nameToken = fieldDeclarator.Identifier;
-                if (nameToken != default)
-                {
-                    name = nameToken.IsMissing ? missingTokenPlaceholder : nameToken.Text;
-                }
+                Debug.Assert(memberDeclaration.Kind() == SyntaxKind.IncompleteMember);
+                return "?";
             }
         }
+        else if (node is VariableDeclaratorSyntax fieldDeclarator)
+        {
+            var nameToken = fieldDeclarator.Identifier;
+            return nameToken.IsMissing ? missingTokenPlaceholder : nameToken.Text;
+        }
 
-        Debug.Assert(name != null, "Unexpected node type " + node.Kind());
-        return name;
-    }
+        Debug.Fail("Unexpected node type " + node.Kind());
+        return null;
 
-    private static void AppendTypeParameterList(StringBuilder builder, TypeParameterListSyntax typeParameterList)
-    {
-        if (typeParameterList != null && typeParameterList.Parameters.Count > 0)
```

Comment:
>typeParameterList.Parameters.Count > 0 [](http://example.com/codeflow?start=41&length=38)

Don't need to worry about Count==0 case any more?

Reply:
not really.  that's an error case anyways.  and in that case, it's fine to say `<>`.